src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1323,25 +1347,33 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
         if (BlockRange().TryGetUse(node, &use))
         {
             GenTree* user = use.User();
+
+            JITDUMP("lowering EmbeddedMasked HWIntrinisic (before):\n");
+            DISPTREERANGE(BlockRange(), node);
+            JITDUMP("\n");
+
             // Wrap the intrinsic in ConditionalSelect only if it is not already inside another ConditionalSelect
-            if (!user->OperIsHWIntrinsic() || (user->AsHWIntrinsic()->GetHWIntrinsicId() != NI_Sve_ConditionalSelect))
-            {
-                CorInfoType simdBaseJitType = node->GetSimdBaseJitType();
-                unsigned    simdSize        = node->GetSimdSize();
-                var_types   simdType        = Compiler::getSIMDTypeForSize(simdSize);
-                GenTree*    trueMask        = comp->gtNewSimdAllTrueMaskNode(simdBaseJitType, simdSize);
-                GenTree*    trueVal         = node;
-                GenTree*    falseVal        = comp->gtNewZeroConNode(simdType);
-
-                GenTreeHWIntrinsic* condSelNode =
-                    comp->gtNewSimdHWIntrinsicNode(simdType, trueMask, trueVal, falseVal, NI_Sve_ConditionalSelect,
-                                                   simdBaseJitType, simdSize);
-
-                BlockRange().InsertBefore(node, trueMask);
-                BlockRange().InsertBefore(node, falseVal);
-                BlockRange().InsertAfter(node, condSelNode);
-                use.ReplaceWith(condSelNode);
-            }
+
+            CorInfoType simdBaseJitType = node->GetSimdBaseJitType();
+            unsigned    simdSize        = node->GetSimdSize();
+            var_types   simdType        = Compiler::getSIMDTypeForSize(simdSize);
+            GenTree*    trueMask        = comp->gtNewSimdAllTrueMaskNode(simdBaseJitType, simdSize);
+            GenTree*    trueVal         = node;
+            GenTree*    falseVal        = comp->gtNewZeroConNode(simdType);
+
+            GenTreeHWIntrinsic* condSelNode =
+                comp->gtNewSimdHWIntrinsicNode(simdType, trueMask, trueVal, falseVal, NI_Sve_ConditionalSelect,
+                                                simdBaseJitType, simdSize);
+
+            BlockRange().InsertBefore(trueVal, trueMask);
```

Comment:
nit: In other places we've been very explicit about doing `create; insert` to ensure it's more obvious about what has been created and that it is definitively inserted.

I actually think using `trueVal` rather than `node` also makes this potentially more confusing, given that `node` is the existing point we're inserting around.

Reply:
> nit: In other places we've been very explicit about doing `create; insert` to ensure it's more obvious about what has been created and that it is definitively inserted.

is there something that I am not doing here? I will change back to using `node` instead of `trueVal`.

Reply:
You're doing all the right things, but you've broken apart where you're creating the node and where you're inserting the node; which deviates from the more typical pattern

Reply:
got it. Will fix.